core/src/js/services/mercenaries/mercenaries-utils.ts

```diff
@@ -215,7 +215,9 @@ export const isPassiveMercsTreasure = (cardId: string, allCards: CardsFacadeServ
 		refCard?.mercenaryPassiveAbility ||
 		refCard.mechanics?.includes(GameTag[GameTag.HIDE_STATS]) ||
 		refCard.mechanics?.includes(GameTag[GameTag.HIDE_COST]) ||
-		refCard.mechanics?.includes(GameTag[GameTag.START_OF_GAME])
+		refCard.mechanics?.includes(GameTag[GameTag.START_OF_GAME]) ||
+		refCard.mechanics?.includes(GameTag[GameTag.DISCOVER]) ||
```

Comment:
DISCOVER doesn't look right here, as you could have an ability that lets you discover something during the game.

Reply:
Yes, I understand, but they don't have speed value displayed in game either. 
Maybe you mean that we need to differentiate the passive treasures from active abilities with hidden speed value (discover abilities)? We could make them appear exactly like in the game - with a speed icon, but without numeric value.

![2022-10-24_18-39-20](https://user-images.githubusercontent.com/43519401/197570277-b53b7060-1218-49ea-a7b4-b8ad1156c193.png)

Reply:
I wasn't aware that such abilities already existed. How does it work then? When you choose Princess Power!, when does it resolve? Immediately (before the orders actually resolve)?

Reply:
There are already three such abilities: Tess' [Princess Power!](https://hearthstone.fandom.com/wiki/Mercenaries/Princess_Power!_5,) Garona's [One And The Same](https://hearthstone.fandom.com/wiki/Mercenaries/One_And_The_Same_5) and Finley' 
 [Murlocology](https://hearthstone.fandom.com/wiki/Mercenaries/Murlocology_5.). It's like abilities with choice of effect, only the speeds of the effects are different.

![2022-10-24_22-07-58](https://user-images.githubusercontent.com/43519401/197608157-08a283c1-f578-4a65-93e4-f496802af6b2.png)
![2022-10-24_22-09-52](https://user-images.githubusercontent.com/43519401/197608164-fff9e5aa-436e-48fa-abed-033a972e9ba6.png)

Reply:
Ok I see, thanks. In that case it's reasonable to assume that this is how all Discover effects will work in the future as well.
So I'm good for now :)

Reply:
It's also necessary to add support for discover abilities to action queue. They now display at 0 speed instead of selected ability speed.

![2022-10-25_14-24-49](https://user-images.githubusercontent.com/43519401/197760727-e6f49a4b-833b-4e4b-8cdb-a64da4a18c01.png)